python/GafferUI/Slider.py

```diff
@@ -46,31 +46,34 @@
 from Qt import QtGui
 from Qt import QtWidgets
 
-## The Slider class allows a user to specify a number of positions on a scale of 0.0 at one end
-# of the Widget and 1.0 at the other. Positions off the ends of the widget are mapped
-# to negative numbers and numbers greater than 1.0 respectively. Derived classes may
-# provide alternative interpretations for the scale and clamp values as appropriate. In
-# particular see the NumericSlider which allows the specification of the values at either
-# end of the scale along with hard minimum and maximum values.
 class Slider( GafferUI.Widget ) :
 
-	PositionChangedReason = IECore.Enum.create( "Invalid", "SetPositions", "Click", "IndexAdded", "IndexRemoved", "DragBegin", "DragMove", "DragEnd", "Increment" )
+	ValueChangedReason = IECore.Enum.create( "Invalid", "SetValues", "Click", "IndexAdded", "IndexRemoved", "DragBegin", "DragMove", "DragEnd", "Increment" )
 
-	def __init__( self, position=None, positions=None, **kw ) :
+	# The min and max arguments define the numeric values at the ends of the slider.
+	# By default, values outside this range will be clamped, but hardMin and hardMax
+	# may be specified to move the point at which the clamping happens outside of the
+	# slider itself.
+	def __init__( self, value=None, min=0, max=1, hardMin=None, hardMax=None, values=None, **kw ) :
 
 		GafferUI.Widget.__init__( self, _Widget(), **kw )
 
-		assert( ( position is None ) or ( positions is None ) )
+		self.__min = min
+		self.__max = max
+		self.__hardMin = hardMin if hardMin is not None else self.__min
+		self.__hardMax = hardMax if hardMax is not None else self.__max
 
-		if positions is not None :
-			self.__positions = positions
+		assert( ( value is None ) or ( values is None ) )
+
+		if values is not None :
+			self.__values = values
 		else :
-			self.__positions = [ 0.5 if position is None else position ]
+			self.__values= [ 0.5 if value is None else value ]
```

Comment:
It's a good question, but I think not. What we're really trying to do here is emulate the default value we would have given to the `value` parameter if it didn't need to be `None` to avoid ambiguity with the `values` parameter. Since your other comment below relates to the confusion of having both `value` and `values`, I've refactored to just have a single `values` argument, with some backwards compatibility for `value` as a keyword argument : 61371aa.